rpc/namespaces/eth/api.go

```diff
@@ -300,17 +326,38 @@ func (api *PublicEthereumAPI) GetBlockTransactionCountByHash(hash common.Hash) *
 	return &n
 }
 
-// GetBlockTransactionCountByNumber returns the number of transactions in the block identified by number.
+// GetBlockTransactionCountByNumber returns the number of transactions in the block identified by its height.
 func (api *PublicEthereumAPI) GetBlockTransactionCountByNumber(blockNum rpctypes.BlockNumber) *hexutil.Uint {
 	api.logger.Debug("eth_getBlockTransactionCountByNumber", "block number", blockNum)
-	height := blockNum.Int64()
-	resBlock, err := api.clientCtx.Client.Block(&height)
+
+	var height int64
+	if blockNum != rpctypes.PendingBlockNumber {
+		if blockNum == rpctypes.LatestBlockNumber {
+			bn, err := api.backend.BlockNumber()
+			if err != nil {
+				return nil
+			}
+			bnCleaned := strings.Replace(bn.String(), "0x", "", -1)
+			height, err = strconv.ParseInt(bnCleaned, 16, 64)
```

Comment:
you can use the util function `api.backend.LatestBlockNumber()`

Reply:
you are 100% right, forgot you move that to backend!